### ISSUE TITLE
When attempting to use other options than "--include-package-data" for data files you have to get it right to be compatible

### ISSUE THREAD

**Issue by marcpaga** [Nuitka/Nuitka#1884]
* Description of the bug:

When including a data directory, one must specify the source directory and the destination directory (where the files will be within the program). This works fine if one uses relative imports in their program. However, if one uses imports that contain the program name `eg. from my_program.module import function` then the provided destination path is not properly handled. 

* Please check out if the develop version of Nuitka works better for you.

  > The bug also happens on the development version.

* Nuitka version, full Python version, flavor, OS, etc. as output by *this*
  command.

  ```
  > python -m nuitka --version
  1.2
  Commercial: None
  Python: 3.7.15 (default, Oct 11 2022, 13:45:04) 
  Flavor: Unknown
  Executable: /Users/mpages/Documents/sturgeon/venv/bin/python3
  OS: Darwin
  Arch: x86_64
  ```

* How did you install Nuitka and Python

  Installed Nuitka via pip on a clean virtual environment.
  ```
  python3 -m venv venv
  pip3 install nuitka
  ```

* The specific PyPI names and versions

  Only Nuitka.

  ```
  > python -m pip freeze
  Nuitka==1.2
  ```

* Also supply a [Short, Self Contained, Correct, Example](https://sscce.org/)

Working setup:
```
my_program
├── include
│   └── models
│       ├── model1.txt
│       └── model2.txt
├── __init__.py
├── main.py
└── utils.py   
```

main.py
```
from utils import find_model_files
find_model_files()
```

utils.py
```
import os

def find_model_files():
    model_path = os.path.join(os.path.dirname(__file__), 'include/models')
    print(model_path)
    for model_file in os.listdir(model_path):
        print(model_file)
    return None
```

Runtime:
```
> ./main.bin
/var/folders/pp/5r_lsyld0s5707v7k1b885h40000gp/T//onefile_69665_1668249666_369330/include/models
model1.txt
model2.txt
```
So the two files can be found by the program at the correct location.

Not working setup:

```
my_program
├── include
│   └── models
│       ├── model1.txt
│       └── model2.txt
├── __init__.py                               
├── main.py
└── utils.py   
```

main.py
```
from my_program.utils import find_model_files # DIFFERENT
find_model_files()
```

utils.py
```
import os

def find_model_files():
    model_path = os.path.join(os.path.dirname(__file__), 'include/models')
    print(model_path)
    for model_file in os.listdir(model_path):
        print(model_file)
    return None
```

Runtime:
```
> ./main.bin
/var/folders/pp/5r_lsyld0s5707v7k1b885h40000gp/T//onefile_70559_1668249846_805317/my_program/include/models
Traceback (most recent call last):
  File "/var/folders/pp/5r_lsyld0s5707v7k1b885h40000gp/T//onefile_70559_1668249846_805317/main.py", line 3, in <module>
  File "/var/folders/pp/5r_lsyld0s5707v7k1b885h40000gp/T//onefile_70559_1668249846_805317/my_program/utils.py", line 7, in find_model_files
FileNotFoundError: [Errno 2] No such file or directory: '/var/folders/pp/5r_lsyld0s5707v7k1b885h40000gp/T//onefile_70559_1668249846_805317/my_program/include/models'
```

So basically, because we used `from my_program.utils import find_model_files` instead of `from utils import find_model_files`, the `utils.py` will now be located at `my_program/utils.py` instead of just `utils.py` and using `os.path.dirname(__file__)` will not point to the expected location.

* Provide in your issue the Nuitka options used

  Nuitka runs fine in building the compiled program. I run the following:
  ```
  python3 -m nuitka \
  --standalone \
  --include-data-dir=my_program/include=include \
  my_program/main.py 
  
  python3 -m nuitka \
  --onefile \
  --include-data-dir=my_program/include=include \
  my_program/main.py 
  ```

**Comment by kayhayen** [comment]
It seems you are including package data with the `include-data-dir` option, do not do that, instead use `--include-package-data` which should spare you of micromanaging filenames, and ought to not do this mistake. It seems you are not putting where the `__file__` suggests, but one level up, which is not matching things. 

Packages have a path relative to the distribution folder, that includes all package names, in your case `my_program`. Adding that to the right side of the `=` part, would also fix it.

However, follow the `--include-package-data=my_program.include.models=*.txt` style of matching, and you will be good. With some use of `**` the top level package could also be sufficient, or without the filename pattern too, since source file types are ignored by default.

**Comment by kayhayen** [comment]
I am kind of considering if we should not warn about using data file options with sources that we can reverse to come from below packages that are also included, as this might well be a common user error. Thinking of a directory, one can be drawn to using this option, which should be simple, but turns out hard to use correctly. Also you should have looked at the dist folder, the difference in file layout should be visible there.
